testutilsint/dinocluster.go

```diff
@@ -85,6 +94,32 @@ func (c *DinoController) cleanup() {
 			c.t.Errorf("failed to reset traffic control for %s", node)
 		}
 	}
+
+	c.EnableAutoFailover()
+}
+
+func (c *DinoController) DisableAutoFailover() {
+	settings, err := getTestMgmt().GetAutoFailoverSettings(context.Background(), &cbmgmtx.GetAutoFailoverSettingsRequest{})
+	require.NoError(c.t, err)
+	c.oldFoSettings = settings
+
+	err = getTestMgmt().ConfigureAutoFailover(context.Background(), &cbmgmtx.ConfigureAutoFailoverRequest{
+		Enabled: ptr.To(false),
+	})
+	require.NoError(c.t, err)
+}
+
+func (c *DinoController) EnableAutoFailover() {
+	if c.oldFoSettings == nil {
+		return
+	}
+
+	err := getTestMgmt().ConfigureAutoFailover(context.Background(), &cbmgmtx.ConfigureAutoFailoverRequest{
+		Enabled: ptr.To(c.oldFoSettings.Enabled),
```

Comment:
Why not just have this as `Enabled: ptr.to(true)`?

Reply:
A bit contrary to the name of the function, this is actually restoring the cluster to whatever state it was in before the test started.  It's exposed as a public function, but it's invoked automatically as part of test cleanup through the hook installed when the DinoController is constructed above.